frontend/server/src/Controllers/Run.php

```diff
@@ -916,6 +946,24 @@ public static function apiSource(\OmegaUp\Request $r): array {
         );
     }
 
+    private static function shouldShowRunDetails(
+        int $identityId,
+        \OmegaUp\DAO\VO\Problems $problem,
+        ?\OmegaUp\DAO\VO\Problemsets $problemset,
+        ?\OmegaUp\DAO\VO\Contests $contest
+    ): bool {
+        if (is_null($problemset) || is_null($problemset->contest_id)) {
+            return \OmegaUp\DAO\Problems::isProblemSolved(
+                $problem,
+                $identityId
+            );
+        }
```

Comment:
este método también debería regresar verdadero si esto es verdadero, aunque L961-L965 hubieran regresado falso. de otra manera alguien que ya resolvió el problema pero no lo está viendo desde el contexto de un concurso no podría ver los detalles ahora!

onda

```suggestion
        if (!is_null($contest) && $contest->feedback !== 'none') {
            return true;
        }
        return \OmegaUp\DAO\Problems::isProblemSolved(
            $problem,
            $identityId
        );
```

también, por qué se está involucrando `$problemset` aquí si ya se tiene acceso a `$contest`?

Reply:
Listo, ya vi que no es necesario el `problemset`.